api/views.py

```diff
@@ -0,0 +1,59 @@
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from rest_framework.viewsets import ModelViewSet
+from rest_framework import status
+
+from . models import Category, Job, Application
+from . serializers import CategorySerializer, JobSerializer, ApplicationSerializer
+
+
+# Create your views here.
+@api_view(['GET'])
+def get_routes(request):
+    routes = [
+        {'GET': '/api/jobs'},
+        {'GET': '/api/application<category_id>'},
+        {'POST': '/api/application'},
+    ]
+
+    return Response(routes)
+
+
+class JobViewSet(ModelViewSet):
+    queryset = Job.objects.all().filter(is_active=True)
+    serializer_class = JobSerializer
+    categories = Category.objects.all()
+
+    def list(self, request):
+        category = request.GET.get('category')
+
+        if category:
+            jobs = self.queryset.filter(category_job__id=category)
+        else:
+            jobs = self.queryset.filter()
+
+        serializer = self.serializer_class(jobs, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class ApplicationViewSet(ModelViewSet):
+    queryset = Application.objects.all()
+    serializer_class = ApplicationSerializer
+    active_jobs = Job.objects.filter(is_active=True)
+    active_jobs_serializer = JobSerializer
+
+    def create(self, request):
+        job_id = request.data['applicant_application']
+        user_email = request.data['email']
+        application = self.queryset.filter(applicant_application_id=job_id, email=user_email).first()
+
+        if not application:
+            serializer = self.serializer_class(data=request.data)
+
+            if serializer.is_valid():
+                serializer.save()
+                return Response(status=status.HTTP_200_OK)
+            return Response(data=serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        return Response(data="already existing", status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Put this in constants.py file and import from there to use.
APPLICATION_ALREADY_SUBMITTED = "You have already submitted the application for this job"

```suggestion
        return Response(data=APPLICATION_ALREADY_SUBMITTED, status=status.HTTP_400_BAD_REQUEST)
```

Reply:
Updated.